Framework/API/inc/MantidAPI/WorkspacePropertyWithIndex.tcc

```diff
@@ -170,21 +173,39 @@ const SpectrumIndexSet WorkspacePropertyWithIndex<TYPE>::getIndices() const {
   if (list.size() == 0)
     return indexInfo.makeIndexSet();
 
-  switch (m_indexTypeProp->selectedType()) {
-  case IndexType::SpectrumNumber:
-    return indexInfo.makeIndexSet(std::vector<Mantid::Indexing::SpectrumNumber>(
-        list.begin(), list.end()));
-    break;
-  case IndexType::WorkspaceIndex:
-    return indexInfo.makeIndexSet(
-        std::vector<Mantid::Indexing::GlobalSpectrumIndex>(list.begin(),
-                                                           list.end()));
-    break;
+  // Determine if input is a pure range or just a list of numbers
+  auto res = std::minmax_element(list.begin(), list.end());
+  auto minIndex = res.first - list.begin();
+  auto maxIndex = res.second - list.begin();
+  bool isPureRange = (list[maxIndex] - list[minIndex] + 1) == list.size();
```

Comment:
Two comments:
- If a user enters "1,2,3,4,5" this will give `isPureRange == true`, even though it isn't? If spectrum 3 does not exist this would work, even though it should not. Maybe this is a corner case that can be ignored, but I think it should be discussed with input from others.
- This is still inefficient for long ranges. Should we maybe consider adding also separate input fields for min and max, such that conversion to a vector can be avoided?